### PR TITLE
feat: more control over no gap for grid layout

### PR DESCRIPTION
**Pull request by stefanoScalzo**
BREAKING CHANGES: `.fd-layout-grid--no-gap` removed
`.fd-layout-grid--no-vertical-gap` added
`.fd-layout-grid--no-horizontal-gap` added
## Related Issue
Closes SAP/fundamental-styles#1803

## Description
This PR adds more control to having no gap for layout grids.
The user can now choose between having no vertical gap which removes space in between rows, no horizontal gap which removes spacing between rows or can have no gap at all.
### Before:
After consists of before and more

### After:
Screen Shot 2020-12-14 at 2 24 43 PM

Screen Shot 2020-12-14 at 2 24 27 PM
Screen Shot 2020-12-14 at 2 24 57 PM

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by JKMarkowski — APPROVED** [review]
LGTM

File: src/fundamental-styles.scss [hunk 1]
```diff
@@ -57,6 +57,7 @@
 @import "./text";
 @import "./textarea";
 @import "./tile";
+@import "./title";
```
Comment (InnaAtanasova) [inline src/fundamental-styles.scss:60]: Why is this added?

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/layout-grid.scss [hunk 2]
```diff
@@ -144,11 +144,18 @@ $cols: 12;
       margin-top: 0;
     }
 
-    &--no-gap {
+    &--no-horizontal-gap {
```
Comment (InnaAtanasova) [inline src/layout-grid.scss:147]: maybe `.row--no-gap` is more suitable (similar to .row-no-gutters in Bootstrap)

File: src/layout-grid.scss [hunk 3]
```diff
@@ -144,11 +144,18 @@ $cols: 12;
...
+      .fd-col {
+        padding-left: 0;
+        padding-right: 0;
+      }
+    }
+
+    &--no-vertical-gap {
```
Comment (InnaAtanasova) [inline src/layout-grid.scss:154]: `.col--no-gap` ?

File: stories/layout-grid/layout-grid.stories.js [hunk 4]
```diff
@@ -290,7 +290,93 @@ offset.parameters = {
     }
 };
 
-export const noGap = () => `<div class="fd-container fd-container--no-gap">
+export const noHorizontalGap = () => `<div class="fd-container fd-container--no-horizontal-gap">
```
Comment (InnaAtanasova) [inline stories/layout-grid/layout-grid.stories.js:293]: Can't we move it to the `fd-row` level?

File: src/fundamental-styles.scss (hunk 1 above)
Comment (stefanoScalzo) [reply src/fundamental-styles.scss]: removed

File: src/layout-grid.scss (hunk 2 above)
Comment (stefanoScalzo) [reply src/layout-grid.scss:147]: As discussed, we can't due to negative margines

File: stories/layout-grid/layout-grid.stories.js (hunk 4 above)
Comment (stefanoScalzo) [reply stories/layout-grid/layout-grid.stories.js:293]: As discussed, we can't due to negative margines

**Review by InnaAtanasova — APPROVED** [review]
